ml-proto/spec/eval.ml

```diff
@@ -294,18 +284,21 @@ and eval_hostop host mem hostop vs at =
     Some (Int32 (Int64.to_int32 (Memory.size mem)))
 
   | GrowMemory, [v] ->
-    let delta = mem_size v at in
+    let delta = address32 v at in
     if I64.rem_u delta host.page_size <> 0L then
-      error at "runtime: grow_memory operand not multiple of page_size";
-    if I64.lt_u (Int64.add (Memory.size mem) delta) (Memory.size mem) then
-      error at "runtime: grow_memory overflow";
-    if mem_overflow (Int64.add (Memory.size mem) delta) then
-      error at "runtime: grow_memory overflow";
+      Trap.error at "growing memory by non-multiple of page size";
+    let new_size = Int64.add (Memory.size mem) delta in
+    if I64.lt_u new_size (Memory.size mem) then
+      Trap.error at "memory size overflow";
+    (* Test whether the new size overflows the memory type.
+     * Since we currently only support i32, just test that. *)
+    if I64.gt_u new_size (Int64.of_int32 Int32.max_int) then
+      Trap.error at "growing memory out of bounds";
```

Comment:
"out of bounds" is a phrase we use to describe accesses that extend beyond the size. How about "memory size exceeded implementation limit"?